plugin/pkg/auth/authorizer/webhook/webhook.go

```diff
@@ -180,10 +185,10 @@ func (w *WebhookAuthorizer) Authorize(attr authorizer.Attributes) (err error) {
 		}
 	}
 	if r.Status.Allowed {
-		return nil
+		return true, "", nil
 	}
 	if r.Status.Reason != "" {
-		return errors.New(r.Status.Reason)
+		return false, r.Status.Reason, nil
```

Comment:
The current implementation checks `Allowed` before `Reason` so I think your example would correctly return allowed.

But yeah, with the current implementation, you can just `return r.StatusAllowed, r.StatusReason, nil`